react/tic-tac-toe/src/app/components/Router/components/CheckRoute/index.js

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import { Redirect, Route } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { string, func } from 'prop-types';
+
+class CheckRoute extends Component {
+  renderRoute = () => {
+    const { component: RouteComponent, path, authToken } = this.props;
+
+    if (authToken) {
+      return path === '/login'
+        ? <Redirect to="/" />
+        : <RouteComponent />;
+    }
+
+    return path === '/login'
+      ? <RouteComponent />
```

Comment:
Is this correct? I'm not sure the logic is correct. If you have an `authToken` and the page ig login you are sending him to a component.

And the else is `if you don't have an authToken and you aren't going to the login then go to the login` 🤔

Reply:
That logic was update, and follow the idea exposed in the meeting with you.